### PR TITLE
Favorite credentials

### PR DESCRIPTION
**Pull request by Devorein**
### WHY

[Card](https://app.charmverse.io/charmverse/set-up-to-5-credentials-as-favourite-9152370013517406)

### CODE REVIEW

File: lib/credentials/external/getExternalCredentials.ts [hunk 1]
```diff
@@ -62,6 +69,57 @@ const GET_EXTERNAL_CREDENTIALS = gql`
   }
 `;
 
+const GET_EXTERNAL_CREDENTIAL = gql`
```
Comment (motechFR) [inline lib/credentials/external/getExternalCredentials.ts:72]: GET_CREDENTIAL

File: lib/credentials/external/getExternalCredentials.ts [hunk 2]
```diff
@@ -62,6 +69,57 @@ const GET_EXTERNAL_CREDENTIALS = gql`
...
+      content: JSON.parse(attestation.decodedDataJson).reduce((acc: any, val: SchemaDecodedItem) => {
+        acc[val.name] = val.value.value;
+        return acc;
+      }, {} as any),
+      timeCreated: attestation.timeCreated * 1000,
+      verificationUrl: getOnChainAttestationUrl({ chainId, attestationId: attestation.id })
+    } as EASAttestationFromApi;
+  } catch (_) {
+    return null;
+  }
```
Comment (motechFR) [inline lib/credentials/external/getExternalCredentials.ts:120]: Lil' cleanup, can we get a reusable utility method in this file please

File: pages/api/credentials/favorites.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  if (!payload.attestationId && !payload.issuedCredentialId) {
+    throw new InvalidInputError('Either attestationId or issuedCredentialId must be provided');
+  }
+  const favoriteCredential = await prisma.favoriteCredential.create({
+    data: {
+      attestationId: payload.attestationId,
+      chainId: payload.chainId,
+      issuedCredentialId: payload.issuedCredentialId
+    }
+  });
```
Comment (motechFR) [inline pages/api/credentials/favorites.ts:34]: Let's make this an upsert instead.

If you define the correct compound indexes on the schema, you'll be able to pass those params to the where clause

See this [comment](https://github.com/charmverse/core/pull/229/files#r1474707880) on core

File: lib/credentials/external/getExternalCredentials.ts (hunk 2 above)
Comment (Devorein) [reply lib/credentials/external/getExternalCredentials.ts]: We got rid of this part of the code, so this mapper is used only in a single place.

**Pull request merged**